error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/IsInstanceLambdaUsage.java

```diff
@@ -39,7 +39,9 @@ public IsInstanceLambdaUsage() {}
 
   @Override
   public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState state) {
-    if (tree.getKind() != Kind.LAMBDA_EXPRESSION || tree.getBody().getKind() != Kind.INSTANCE_OF) {
+    if (tree.getKind() != Kind.LAMBDA_EXPRESSION
+        || tree.getBody().getKind() != Kind.INSTANCE_OF
```

Comment:
It looks like `tree.getBody().getKind() == Kind.INSTANCE_OF` implies `tree.getKind() == Kind.LAMBDA_EXPRESSION`, so we can skip the latter check.

Reply:
I was looking at this but I think this is not correct. Isn't it because we are implementing a `LambdaExpressionTreeMatcher`, that makes the first condition always true?

Reply:
^ Discussed offline: in case of a "block lambda" we do hit this code, but `tree.getBody().getKind()` will then be `    BLOCK`.